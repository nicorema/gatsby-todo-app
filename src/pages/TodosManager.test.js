import React from 'react'
import { Provider } from 'react-redux'
import renderer from 'react-test-renderer'
import configureStore from 'redux-mock-store'

import TodosManager from './todos-manager'

const mockStore = configureStore([])

describe('Todos Manager Page', () => {
  let store
  let todoManagerCmp

  beforeEach(() => {
    store = mockStore({
      app: { todos: [] },
    })
    todoManagerCmp = renderer.create(
      <Provider store={store}>
        <TodosManager />
      </Provider>
    )
  })

  it('renders <TodoList />', () => {
    const snapShot = todoManagerCmp.toJSON()
    expect(snapShot).toMatchSnapshot()
  })

  it('renders <TodoList />', () => {
    const list = todoManagerCmp.root.findByProps({ 'data-testid': 'list' })
    expect(list).toBeTruthy()
  })
})
