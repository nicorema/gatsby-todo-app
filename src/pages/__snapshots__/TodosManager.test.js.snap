// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Todos Manager Page renders <TodoList /> 1`] = `
<div
  className="css-1mjppp"
>
  <header
    className="sc-dlnjPT eSGkCO"
  >
    <h1>
      todos.manager.title
    </h1>
    <div
      className="sc-gtssRu loStDX"
    >
      <button
        onClick={[Function]}
      >
        todos.manager.create.label
      </button>
    </div>
  </header>
  <ul
    className="sc-gKAblj eekmid"
  />
  <footer
    className="css-1ty9u91"
  >
    <nav
      className="sc-bdnylx giuwNz"
    >
      <ul>
        <li>
          <span
            className="css-qyfojd"
          >
            <a
              href="/en-US/todos-manager"
              onClick={[Function]}
            >
              Todos Manager
            </a>
          </span>
        </li>
        <li>
          <span
            className="css-qyfojd"
          >
            <a
              href="/en-US/todos-log"
              onClick={[Function]}
            >
              Todos Log
            </a>
          </span>
        </li>
      </ul>
    </nav>
  </footer>
</div>
`;
